esmarlint/CorePagination
# Feature/benchmarks tests

# Benchmarks and Code Organization

## Description
This pull request introduces comprehensive benchmarks for various components of CorePagination and improves the organization and structure of the benchmark code. The main changes include:

- Added benchmarks for `PaginatorExtensions` methods:
  - `PaginateAsync`
  - `SimplePaginateAsync`
  - `CursorPaginateAsync`
  - `Paginate`
  - `SimplePaginate`
  - `CursorPaginate`

- Added benchmarks for `Paginator` classes:
  - `SizeAwarePaginator`
  - `SimplePaginator`
  - `CursorPaginator`

- Added benchmarks for `Transformer` classes:
  - `SizeAwareUrlResultTransformer`
  - `SimpleUrlResultTransformer`
  - `CursorUrlResultTransformer`

- Added benchmarks for `TransformerExtensions` methods:
  - `WithUrl` for `SizeAwarePaginator`
  - `WithSimpleUrl` for `SimplePaginator`
  - `WithUrl` for `CursorPaginator`

- Organized benchmark classes into regions for better code structure and readability:
  - `PaginatorExtensionsBenchmarks` now has regions for asynchronous and synchronous pagination benchmarks.
  - `PaginatorBenchmarks` now has regions for each paginator type (`SizeAwarePaginator`, `SimplePaginator`, `CursorPaginator`).

- Updated benchmark methods to use a consistent naming convention:
  - Asynchronous methods end with `Async` suffix.
  - Synchronous methods have no suffix.

- Fixed missing benchmark for `CursorPaginate` method in `PaginatorExtensionsBenchmarks`.

## Motivation
The purpose of this pull request is to enhance the benchmark coverage of CorePagination and provide a more organized and readable structure for the benchmark code. By adding comprehensive benchmarks for various components, we can better assess the performance characteristics of CorePagination and identify potential areas for optimization.

The introduction of regions and consistent naming conventions improves the overall code organization and makes it easier for developers to navigate and understand the benchmark code.

## Testing
To verify the changes, follow these steps:
1. Pull the latest changes from the branch.
2. Build the CorePagination solution.
3. Navigate to the `CorePagination.Benchmarks` project.
4. Run the benchmark tests using your preferred benchmark runner (e.g., BenchmarkDotNet).
5. Review the benchmark results and ensure that all benchmarks execute successfully.

## Checklist
- [x] Added benchmarks for `PaginatorExtensions` methods
- [x] Added benchmarks for `Paginator` classes
- [x] Added benchmarks for `Transformer` classes
- [x] Added benchmarks for `TransformerExtensions` methods
- [x] Organized benchmark code into regions
- [x] Updated benchmark method naming convention
- [x] Fixed missing benchmark for `CursorPaginate` method
- [x] Tested benchmark execution and verified results

Please review the changes and provide any feedback or suggestions. Let me know if you have any questions!
